api/tag/janitor_test.go

```diff
@@ -14,12 +16,109 @@ func TestJanitor(t *testing.T) {
 	tagStore := NewMemoryStore()
 	taskProvider := mock_provider.NewMockTaskProvider(ctrl)
 
-	// todo: put some old tags in the store, mock the ListTasks call
+	tags := models.Tags{
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn2",
+		},
+		{
+			EntityID:   "env_id1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name1",
+		},
+		{
+			EntityID:   "env_id2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	taskSummaries := []models.TaskSummary{
+		{
+			TaskID:          "tsk_id1",
+			TaskName:        "tsk_name1",
+			EnvironmentID:   "env_id1",
+			EnvironmentName: "env_name1",
+		},
+		{
+			TaskID:          "tsk_id2",
+			TaskName:        "tsk_name2",
+			EnvironmentID:   "env_id2",
+			EnvironmentName: "env_name2",
+		},
+	}
+
+	taskProvider.
+		EXPECT().
+		List().
+		Return(taskSummaries, nil)
+
+	actual, _ := taskProvider.List()
+
+	taskProvider.
+		EXPECT().
+		List().
+		Return(taskSummaries, nil)
 
 	janitor := NewJanitor(tagStore, taskProvider)
 	if err := janitor.Run(); err != nil {
 		t.Fatal(err)
 	}
 
-	// todo: assert only old tags got deleted
+	expected := []models.TaskSummary{
```

Comment:
This doesn't make any sense - you're only asserting that the mock call on line `96` returned the objects you told it to return on line `91`. There aren't any expected calls anywhere that would verify that delete was called to any leftover tags.